web/src/js/__tests__/ducks/modes/localSpec.tsx

```diff
@@ -0,0 +1,148 @@
+import localReducer, {
+    toggleLocal,
+    setApplications,
+    initialState,
+    getMode,
+    sanitizeInput,
+} from "../../../ducks/modes/local";
+import { toggleRegular } from "../../../ducks/modes/regular";
+import * as options from "../../../ducks/options";
+import { TStore } from "../tutils";
+import fetchMock, { enableFetchMocks } from "jest-fetch-mock";
+
+describe("localReducer", () => {
+    it("should return the initial state", () => {
+        const state = localReducer(undefined, {});
+        expect(state).toEqual(initialState);
+    });
+
+    it("should dispatch MODE_LOCAL_TOGGLE and updateMode", async () => {
+        enableFetchMocks();
+
+        const store = TStore();
+
+        expect(store.getState().modes.local.active).toBe(false);
+        await store.dispatch(toggleLocal());
+        expect(store.getState().modes.local.active).toBe(true);
+        expect(fetchMock).toHaveBeenCalled();
+    });
+
+    it("should dispatch MODE_LOCAL_SET_APPLICATIONS and updateMode", async () => {
+        enableFetchMocks();
+        const store = TStore();
+
+        await store.dispatch(setApplications("curl"));
+
+        const state = store.getState().modes.local;
+        expect(state.applications).toEqual("curl");
+        expect(fetchMock).toHaveBeenCalled();
+    });
+
+    it('should handle UPDATE_OPTIONS action with data.mode not containing "local"', async () => {
+        const store = TStore();
+
+        await store.dispatch(setApplications("curl"));
+
+        await store.dispatch(toggleRegular());
+
+        expect(store.getState().modes.local.active).toBe(false);
+        expect(store.getState().modes.regular.active).toBe(false);
+        expect(store.getState().modes.local.applications).toBe("curl");
+    });
+
+    it('should handle RECEIVE_OPTIONS action with data.mode containing "local" and an application', () => {
+        const action = {
+            type: options.RECEIVE,
+            data: {
+                mode: {
+                    value: ["local:curl"],
+                },
+            },
+        };
+        const newState = localReducer(undefined, action);
+        expect(newState.active).toBe(true);
+        expect(newState.applications).toBe("curl");
+        expect(newState.error).toBeUndefined();
+    });
+
+    it('should handle RECEIVE_OPTIONS action with data.mode containing "local"', () => {
+        const initialState = {
+            active: false,
+            name: "local",
+            applications: "curl",
+        };
+        const action = {
+            type: options.RECEIVE,
+            data: {
+                mode: {
+                    value: ["local"],
+                },
+            },
+        };
+        const newState = localReducer(initialState, action);
+        expect(newState.active).toBe(true);
+        expect(newState.applications).toBe("curl");
```

Comment:
Great we have this in a test now! But this here looks wrong? UI would show "curl" whereas backend ingests everything without filter.

Reply:
This was a question I added above:

Also related to this and to the local mode:
Let's say we write curl in the applications field in the Capture tab, without activating the local mode. Then we go in the options modal and we write local. Should we activate local:curl or just local? At the moment it's the second case. Same thing for the regular mode with the listen_host and the listen_port.

Reply:
Answered above, sorry for missing that thread.